GetandTake/Core/DataAccess/BaseRepository.cs

```diff
@@ -19,42 +18,88 @@ public BaseRepository(TContext dbContext)
         _entities = _dbContext.Set<TEntity>();
     }
 
-    public IQueryable<TEntity> AsNoTracking()
+    public async Task<List<TEntity>> GetAllItemsAsync() =>
+                await _entities.AsNoTracking().ToListAsync();
+
+    public async Task<List<TEntity>> GetItemsByFilterWithIncludesAsync(
+               Expression<Func<TEntity, bool>> expression,
+               params Expression<Func<TEntity, object>>[] includes)
     {
-        return _entities.AsNoTracking();
+        var query = _entities.Where(expression);
+
+        if (includes.Any())
+        {
+            query = includes
+                .Aggregate(query,
+                    (
+                        current, includeProperty) => current.Include(includeProperty)
+                );
+        }
+
+        return await query.ToListAsync();
+    }
+
+    public async Task<List<TEntity>> GetItemsWithIncludesAsync(params Expression<Func<TEntity, object>>[] includes)
+    {
+        var query = _entities.AsNoTracking();
+        if (includes.Any())
+        {
+            query = includes
+                .Aggregate(query,
+                    (
+                        current, includeProperty) => current.Include(includeProperty)
+                );
+        }
+
+        return await query.ToListAsync();
     }
 
     public async Task CreateAsync(TEntity entity)
     {
         await _entities.AddAsync(entity);
+        _dbContext.SaveChanges();
     }
 
     public void Delete(Expression<Func<TEntity, bool>> expression)
     {
         _entities.RemoveRange(_entities.Where(expression));
+        _dbContext.SaveChanges();
     }
 
-    public async Task<TEntity> GetAsync(Expression<Func<TEntity, bool>> filter)
+    public async Task<TEntity> GetAsync(Expression<Func<TEntity, bool>> filter, params Expression<Func<TEntity, object>>[] includes)
     {
         try
         {
             var query = _entities.Where(filter).AsNoTracking();
-            var item = await query.SingleOrDefaultAsync();
-            return item;
+            if (includes.Any())
+            {
+                query = includes
+                    .Aggregate(query,
+                        (
+                            current, includeProperty) => current.Include(includeProperty)
+                    );
+            }
+            return await query.SingleOrDefaultAsync();
         }
         catch (Exception ex)
         {
             throw new Exception($"Unable to find item in database. Error: {ex.Message}");
         }
     }
-   
-    public List<TEntity> GetAll()
-    {
-        return _dbContext.Set<TEntity>().ToList();
-    }
 
     public void Update(TEntity entity)
     {
         _entities.Update(entity);
+        _dbContext.SaveChanges();
+    }
+
+    public async Task<List<TEntity>> GetItemsByLimit(int limit)
+    {
+        List<TEntity> items;
+        items = await _entities
+            .Take(limit)
+            .AsNoTracking()
+            .ToListAsync();
+        return items;
```

Comment:
You may just put it in single line:
```suggestion
        public async Task<List<TEntity>> GetItemsByLimit(int limit) =>
              await _entities
                  .Take(limit)
                  .AsNoTracking()
                  .ToListAsync();
```